GitForWindowsHelper/index.js

```diff
@@ -31,6 +31,19 @@ module.exports = async function (context, req) {
         return withStatus(500, undefined, e.toString('utf-8'))
     }
 
+    try {
+        const selfHostedARM64Runners = require('./self-hosted-arm64-runners')
+        if (req.headers['x-github-event'] === 'workflow_job'
+            && req.body.repository.full_name === 'git-for-windows/git-for-windows-automation'
+            && ['queued', 'completed'].includes(req.body.action)
+            && req.body.workflow_job.labels.length === 2
+            && req.body.workflow_job.labels[0] === 'Windows'
+            && req.body.workflow_job.labels[1] === 'ARM64') return ok(await selfHostedARM64Runners(context, req))
+    } catch (e) {
+        context.log(e)
+        return withStatus(500, undefined, e.toString('utf-8'))
```

Comment:
Is "500" the right status code here? If someone happens to request this from a repo that's not `git-for-windows/git-for-windows-automation`, it should be a `400` instead, right? Otherwise GitHub might retry the same request multiple times. At least that's what I'm used to based on other webhook implementations I did in the past. It looks like GitHub [doesn't retry requests at all](https://github.com/orgs/community/discussions/24721#discussioncomment-3245226), though it's not official documentation.

WDYT?

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses, a 400 might be interpreted as somehow something being wrong with the webhook event delivery, but the problem in that instance lies with the original actor to which the webhook event was just a response.

In other words, I fear that anything but 500 might be misinterpreted as either retry-able or worse, as "don't ever send anything there again, the server is broken".

That's why the code is already littered with responses using the status code 500 to indicate that the App wants to report an error.